src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Cache.cs

```diff
@@ -76,8 +74,9 @@ public static int CacheSize
         {
             // to avoid lock:
             CachedCodeEntry? first = s_cacheFirst;
-            if (first?.Key == key)
+            if (first != null && first.Key.Equals(key))
```

Comment:
@stephentoub - this will result in a NRE if `first.Key` is `null`

Reply:
@udlose, Key is a struct; it can't be null.

Reply:
That's what I get for not looking at the codeine it's full context. Sorry for the wasted bits.